template/core/data/src/main/java/io/bloco/core/data/network/OpenLibraryService.kt

```diff
@@ -1,34 +1,40 @@
 package io.bloco.core.data.network
 
-import io.bloco.core.data.models.Book
+import io.bloco.core.data.models.BookDetailsDto
 import io.bloco.core.data.models.BookRecords
-import io.ktor.client.HttpClient
 import io.ktor.client.call.body
 import io.ktor.client.request.get
 import io.ktor.client.request.url
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
 import javax.inject.Inject
 
 class OpenLibraryService
 @Inject constructor(
-    @OpenLibraryHttpClient private var httpClient: HttpClient
+    private val httpClient: OpenLibraryHttpClient
 ) {
-    suspend fun getBooks(): Result<BookRecords> = try {
-        Result.success(
-            httpClient.get {
-                url(path = "/")
-            }.body()
-        )
-    } catch (e: Exception) {
-        Result.failure(e)
+
+    suspend fun getBooks(): Result<BookRecords> = withContext(Dispatchers.IO) {
+        return@withContext try {
+            Result.success(
+                httpClient().get {
+                    url(path = "/search.json?q=android&limit=10")
+                }.body()
+            )
+        } catch (e: Exception) {
```

Comment:
Is there a more specific exception?

Reply:
I think this Exception is to cover the cast of the requested result body()... 
so I guess we can have it more specific by validation the status code like:

```
when (response.status) {
            HttpStatusCode.BadRequest -> {} // Throw errors or transform to T 
            HttpStatusCode.Conflict -> {}
            else -> throw this
        }
```

and/or only attempt to catch the cast exception.

but @LilianaFaustinoBloco developed this part, can you confirm me this?

Reply:
Yep, we can do this. 

But to handle this type of exception, I think we need a [HttpResponseValidator](https://ktor.io/docs/response-validation.html#non-2xx).